### PR TITLE
Vector Distance Crash Tests Grammar

### PR DESCRIPTION
**Pull request by saikumar-vs**
- Introduced `init_vectors.sql` to correct vector dimensions and ensure all required IDs are represented for testing.
- Added `README.md` to provide guidance on the usage and structure of the new grammar files.
- Created `vector_distance.yy` for extended RQG grammar, covering various error paths and valid queries for VECTOR_DISTANCE.
- Included `vector_distance.zz` as a companion gendata file to support the new grammar.

Steps to Test : perl runall-new.pl --post-gendata-sql=conf/vector_distance/init_vectors.sql --basedir=<basedir-path>  --grammar=conf/vector_distance/vector_distance.yy   --gendata=conf/vector_distance/vector_distance.zz   --threads=4   --queries=50000   --duration=600  --reporter=Shutdown,Backtrace,ErrorLog,QueryTimeout --debug --sqltrace

### CODE REVIEW

**Review by vinperothas — COMMENTED** [review]
Hi Sai, Thanks for putting this together! The RQG grammar coverage for the VECTOR_DISTANCE edge cases looks really solid, and I appreciate the thoroughness of the boundary tests.

I did notice that the README.md and some of the block comments in init_vectors.sql (like the notes about 'our manual testing session' and 'your actual checkout') read a bit like a conversational chat rather than standard documentation.

Could we polish the README and the SQL comments to be more objective? Stripping out the conversational framing will make it a much cleaner guide

File: randgen/conf/vector_distance/vector_distance.yy [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+_metric_prefix:
+	'COS' | 'EUC' | 'MAN' ;
+
+_metric_suffix:
+	'INE' | 'LIDEAN' | 'HATTAN' ;
+
+_bad_byte_literal:
+	0x0102030405 | 0x010203 | 0x0102030405060708090A | 0x01 ;
+
+# Base Standard Vector Generators
```
Comment (vinperothas) [inline randgen/conf/vector_distance/vector_distance.yy:277]: Impressive use of inline perl 👍

File: randgen/conf/vector_distance/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+## Syntax basis
+
+This grammar's syntax was aligned against `conf/examples/example.yy` from your actual checkout, which confirmed:
+- Rules are semicolon-terminated (`... ;`)
+- Alternatives are `|`-separated, and may span multiple lines
+- Underscore-prefixed tokens (`_table`, `_field`, `_digit`) are RQG's built-in generators, tied to whatever `--gendata` builds
+
+Since this grammar needs `VECTOR`-typed columns (which `--gendata`'s built-in generator doesn't produce), it defines its own literal table (`vt`) with explicit columns in `query_init`, rather than using `_table`/`_field`. This is consistent with how `example.yy` itself falls back to literal SQL identifiers where the built-in generators don't fit.
```
Comment (vinperothas) [inline randgen/conf/vector_distance/README.md]: Shall be updated that the schema is created inside init_vectors.sql

**Comment by saikumar-vs** [comment]
> Hi Sai, Thanks for putting this together! The RQG grammar coverage for the VECTOR_DISTANCE edge cases looks really solid, and I appreciate the thoroughness of the boundary tests.
> 
> I did notice that the README.md and some of the block comments in init_vectors.sql (like the notes about 'our manual testing session' and 'your actual checkout') read a bit like a conversational chat rather than standard documentation.
> 
> Could we polish the README and the SQL comments to be more objective? Stripping out the conversational framing will make it a much cleaner guide

Thanks for feedback Vin, i have addressed this request please take a look

**Review by vinperothas — APPROVED** [review]
